packages/widget-chat/src/reducers/notifications.js

```diff
@@ -0,0 +1,37 @@
+import {ADD_NOTIFICATION, MARK_NOTIFICATION_SENT, UPDATE_NOTIFICATION_SETTING} from '../actions/notifications';
+
+export default function indicators(state = {
+  isSupported: false,
+  items: [],
+  permission: null
+}, action) {
+  switch (action.type) {
+  case ADD_NOTIFICATION:
+    {
+      const {notificationId, type} = action.payload.notification;
+      if (state.items.find((notification) => notification.notificationId === notificationId)) {
+        // Don't add notifications for items already added
+        return state;
+      }
+      const notification = {
+        notificationId,
+        type,
+        sent: false
+      };
+      return Object.assign({}, state, {items: [notification, ...state.items]});
+    }
+  case MARK_NOTIFICATION_SENT:
+    return Object.assign({}, state, {
+      items: state.items.map((notification) => {
+        if (notification.notificationId === action.payload.notificationId) {
+          notification.sent = true;
+        }
+        return notification;
+      })
+    });
+  case UPDATE_NOTIFICATION_SETTING:
+    return Object.assign({}, state, action.payload.setting);
```

Comment:
This feels a bit unsafe since anything passed as a setting payload can overwrite anything in the state. Maybe we should create a settings object inside the main store for these sorts of things?